### PR TITLE
Fix speech to text scripts.

### PR DESCRIPTION
**Pull request by VahidooX**
We need to use the trainer object, no the config. It forces the user to have those in the config file.
Also fixed the check_slurm method to detect the slurm.

### CODE REVIEW

**Review by titu1994 — APPROVED** [review]
Makes sense, LGTM

File: nemo/utils/exp_manager.py [hunk 1]
```diff
@@ -473,7 +473,7 @@ def get_log_dir(
                 logging.warning(
                     "No version folders would be created under the log folder as 'resume_if_exists' is enabled."
                 )
-                version = ""
+                version = None
```
Comment (titu1994) [inline nemo/utils/exp_manager.py:476]: Ah this should have remained "". I'll fix this in another pr

Comment (VahidooX) [reply nemo/utils/exp_manager.py:476]: I was testing something. It works without error now. Let me try something before reverting it back. My suspicion is that w&b handler methods think we want to use version even when we pass "" as the version.

Comment (titu1994) [reply nemo/utils/exp_manager.py:476]: Ok I'll leave that to you then, I haven't attempted the wandb fox yet.

**Pull request merged**